File: modules/server-play-sample/app/controllers/Event.scala
package controllers

import javax.xml.bind.annotation.{XmlAccessType, XmlAccessorType, XmlRootElement}

import be.wegenenverkeer.atomium.play.jaxb.Adapters.StringOptionAdapter
import be.wegenenverkeer.atomium.play.jaxb.{xmlAttribute, xmlElement, xmlTypeAdapter}

import play.api.libs.json._

object EventFormat {
  implicit val eventFormat = Json.format[Event]
}

@XmlRootElement
@XmlAccessorType (XmlAccessType.NONE)
case class Event(@xmlElement value: Double,
                 @xmlElement @xmlTypeAdapter(classOf[StringOptionAdapter]) description: Option[String],
                 @xmlAttribute version: Int = 1) {
    def this() = this(0.0, None, 0)

}

File: modules/client-java/src/test/java/be/wegenenverkeer/atomium/client/Event.java
package be.wegenenverkeer.atomium.client;

import javax.xml.bind.annotation.*;

// this is the Event model, annotations for XML deserialization
// and public accessors for JSON
@XmlRootElement
@XmlAccessorType(XmlAccessType.NONE)
public class Event {

    @XmlElement
    public Double value;

    @XmlElement
    public String description;

    @XmlAttribute
    public Integer version;

    public Event(){
    }


    public String toString() {
        return "Event " + version + " " + "description " + " value: " + value;
    }

}
